classes/bodyText/Repository.php

```diff
@@ -0,0 +1,198 @@
+ * @class Repository
+ *
+ * @brief A repository to find and manage body text files.
+ */
+
+namespace PKP\bodyText;
+
+use APP\core\Application;
+use APP\facades\Repo;
+use APP\publication\Publication;
+use APP\submission\Submission;
+use Exception;
+use PKP\context\Context;
+use PKP\file\TemporaryFileManager;
+use PKP\submissionFile\SubmissionFile;
+use PKP\user\User;
+
+class Repository
+{
+    /**
+     * Map the BodyTextFile to an array including body text content and dependent files
+     */
+    public function map(BodyTextFile $bodyTextFile): array
+    {
+        $fileProps = [];
+
+        if ($bodyTextFile->submissionFile) {
+            $submission = Repo::submission()->get($bodyTextFile->submissionId);
+            $schemaMap = Repo::submissionFile()->getSchemaMap($submission, []);
+
+            $fileProps = $schemaMap->map($bodyTextFile->submissionFile);
+        }
+
+        if ($bodyTextFile->bodyTextContent) {
+            $fileProps['bodyTextContent'] = $bodyTextFile->bodyTextContent;
+        }
+
+        if ($bodyTextFile->loadingContentError) {
+            $fileProps['loadingContentError'] = $bodyTextFile->loadingContentError;
+        }
+
+        return $fileProps;
+    }
+
+    /**
+     * Returns the SubmissionFile, if any, that corresponds to the body text contents of the given submission/publication
+     */
+    public function getBodyTextFile(int $publicationId, ?int $submissionId = null): BodyTextFile
+    {
+        $submissionFileQuery = Repo::submissionFile()
+            ->getCollector()
+            ->filterByFileStages([SubmissionFile::SUBMISSION_FILE_BODY_TEXT])
+            ->filterByAssoc(Application::ASSOC_TYPE_PUBLICATION, [$publicationId]);
+
+        if ($submissionId) {
+            $submissionFileQuery = $submissionFileQuery->filterBySubmissionIds([$submissionId]);
+        }
+
+        $submissionFile = $submissionFileQuery
+            ->getMany()
+            ->first();
+
+        return new BodyTextFile(
+            $publicationId,
+            $submissionId,
+            $submissionFile,
+        );
+    }
+
+    /**
+     * Create or update the body text file for a publication
+     */
+    public function setBodyText(
+        string $bodyText,
+        int $publicationId,
+        ?int $submissionId = null,
+    ): BodyTextFile {
+        $publication = Repo::publication()->get($publicationId, $submissionId);
+        $submission = Repo::submission()->get($publication->getData('submissionId'));
+
+        $existingBodyTextFile = $this->getBodyTextFile($publicationId, $submission->getId());
+
+        if ($existingBodyTextFile->submissionFile) {
+            return $this->updateBodyTextFile($existingBodyTextFile->submissionFile, $bodyText, $submission);
+        }
+
+        $context = Application::get()->getRequest()->getContext();
+        $user = Application::get()->getRequest()->getUser();
+
+        return $this->createBodyTextFile($bodyText, $publication, $submission, $context, $user);
```

Comment:
The thrown exceptions in this method don't look like they're caught anywhere here. Is that on purpose or should they be caught here or at the API controller level?